### PR TITLE
feat: Add solidity module.

### LINKED ISSUE #3869: Add Solidity as language
## Feature Request
Add Solidity language, Solidity is the language most used for developing smart contracts on multiple blockchains.

#### Describe the solution you'd like
Add Checks for when .sol files are in the project so we can create plugins based on Solidity.

### PR DESCRIPTION
**Pull request by aciba90**
#### Description

- The `solidity` module shows the currently installed version of [Solidity](https://soliditylang.org/)
- The module will be shown if any of the following conditions are met:
      - The current directory contains a file with the `.sol` extension

#### Motivation and Context

Closes #3869

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

File: src/modules/solidity.rs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            log::warn!("Error in module `solidity`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn get_module_version(context: &Context, _config: &SolidityConfig) -> Option<String> {
+    let version = parse_solc_version(&context.exec_cmd("solc", &["--version"])?.stdout)?;
```
Comment (davidkna) [inline src/modules/solidity.rs:59]: Should this also support `solcjs`?

**Review by davidkna — COMMENTED** [review]
The changes mostly look good. Please also update the presets in `docs/.vuepress/public/presets/toml`.

**Comment by davidkna** [comment]
@aciba90 Please reply to the comments I posted.

**Comment by AnirbanHalder654322** [comment]
Hello , i have noticed that this hasn't been touched for a quite while . Would anyone mind if i start working on it ?

**Pull request closed without merging**